#!/usr/bin/env python
PACKAGE = "gpt4v_vqa"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_height", int_t, 0, "Maximum image height", 480, 0, 1080)
gen.add("max_width", int_t, 0, "Maximum image width", 640, 0, 1920)
gen.add("detail_level", str_t, 0, "Detail level of GPT4V API", "low")

exit(gen.generate(PACKAGE, "gpt4v_vqa", "GPT4V"))